conda_forge_tick/chained_db.py

```diff
@@ -0,0 +1,72 @@
+"""Base class for chaining DBs
+
+ChainDBSingleton
+Copyright 2015-2016, the xonsh developers
+"""
+
+import itertools
+
+from collections import ChainMap
+from collections.abc import MutableMapping
+
+
+class ChainDBSingleton(object):
+    """Singleton for representing when no default value is given."""
+
+    __inst = None
+
+    def __new__(cls):
+        if ChainDBSingleton.__inst is None:
+            ChainDBSingleton.__inst = object.__new__(cls)
+        return ChainDBSingleton.__inst
+
+
+Singleton = ChainDBSingleton()
```

Comment:
I don't like calling this a Singleton.  I would recommend following what None and NoneType do.  So this would be something like:

```
ChainDBDefault = ChainDBDefaultType()
```

Reply:
Also, is this something that should go into `xonsh.lib`?

Reply:
This is code vendorized from regolith. It could go into `xonsh.lib`

Reply:
Right, so rather than copy it up the stack, we could copy it down :)

Reply:
xref: https://github.com/xonsh/xonsh/pull/2809